actioncable/lib/assets/javascripts/action_cable/connection.coffee

```diff
@@ -60,11 +60,15 @@ class ActionCable.Connection
       switch type
         when message_types.confirmation
           @consumer.subscriptions.notify(identifier, "connected")
+          @dispatchEvent("connected", message, JSON.parse(identifier).channel)
         when message_types.rejection
           @consumer.subscriptions.reject(identifier)
         else
           @consumer.subscriptions.notify(identifier, "received", message)
 
+          if identifier isnt ActionCable.INTERNAL.identifiers.ping
+            @dispatchEvent("received", message, JSON.parse(identifier).channel)
```

Comment:
Took me a while, but we have to parse the identifier because it's stringified when creating the subscription.

Think we should just have `@dispatchEvent` act on an identifier like `subscriptions.notify`. Also don't think we should default the parameters to null anymore (since we've removed all the instances where they are passed as null now). Here's my thinking:

```coffeescript
dispatchEvent: (identifier, eventName, message) ->
  channel = JSON.parse(identifier).channel

  event = document.createEvent('Event')
  event.initEvent("actioncable:#{eventName}", true, false)
  event.data = {message, channel}
  document.dispatchEvent(event)

```